core/components/broker/broker.go

```diff
@@ -270,6 +279,12 @@ func (b component) HandleData(bctx context.Context, req *core.DataBrokerReq) (*c
 		}
 
 		if ok {
+			if counterReset {
+				if !entry.DevMode {
+					continue
```

Comment:
Eeehm, don't you think I tested this code with a life node and gateway? If a reset is 'detected' the counter will be set to the value as provided by the frame, the MIC check will succeed and the value of ok will be true. So if the counter has been reset and the device is not set to development mode the value gets rejected as it might be a failure of the device or a replay attack.